client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -1,13 +1,65 @@
 import json
 import math
 import os
-
+from typing import List
 from ayon_api import get_representation_by_id
 from ayon_maya.api import plugin
 from maya import cmds
 from maya.api import OpenMaya as om
 
 
+def convert_matrix_to_4x4_list(
+        value) -> List[List[float]]:
+    """Convert matrix or flat list to 4x4 matrix list
+    Example:
+        >>> convert_matrix_to_4x4_list(om.MMatrix())
+        [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]
+        >>> convert_matrix_to_4x4_list(
+        ... [1, 0, 0, 0,
+        ...  0, 1, 0, 0,
+        ...  0, 0, 1, 0,
+        ...  0, 0, 0, 1]
+        ... )
+        [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]
+    """
+    result = []
+    value = list(value)
+    for i in range(0, len(value), 4):
+        result.append(list(value[i:i + 4]))
+    return result
+
+
+def convert_transformation_matrix(transform_mm: om.MMatrix, rotation: list) -> om.MMatrix:
+    """Convert matrix to list of transformation matrix for Unreal Engine import.
+
+    Args:
+        transform_mm (om.MMatrix): World Matrix for the asset
+        rotation (list): Rotations of the asset
+
+    Returns:
+        List[om.MMatrix]: List of transformation matrix of the asset
+    """
+    convert_transform = om.MTransformationMatrix(transform_mm)
+
+    convert_tanslation = convert_transform.translation(om.MSpace.kWorld)
+    convert_tanslation = om.MVector(convert_tanslation.x, convert_tanslation.z, -convert_tanslation.y)
+    convert_scale = convert_transform.scale(om.MSpace.kObject)
+    convert_transform.setTranslation(convert_tanslation, om.MSpace.kWorld)
+    if math.radians(rotation[0]) <= 0 or math.radians(rotation[0]) == 0 and math.radians(rotation[1]) == 0:
```

Comment:
The radians conversion here is redundant. We're comparing against 0 and the conversion of radians/degrees would make no difference to that. 

Also, the second check "looks redundant" because it appears included in the first check BUT instead due to `or`  versus `and` ordering I have a feeling that last `and` is merged with the second `or`. However, I'd still add parentheses then for clarity.
```suggestion
    if rotation[0] <= 0 or (rotation[0] == 0 and rotation[1] == 0):
```

Also, HOW did you come up with this solution. This feels overly "magical" and by pure chance just applying certain adjusted values. It smells like something that magically works for one case but will bite us for other cases. Would be great if a comment could be added explaining WHY this works so future us understand WHY it's done this way.

Reply:
> Also, HOW did you come up with this solution. This feels overly "magical" and by pure chance just applying certain adjusted values. It smells like something that magically works for one case but will bite us for other cases. Would be great if a comment could be added explaining WHY this works so future us understand WHY it's done this way.

I hope I can explain why in certain way, but for certain case, it would be even less persuasive with the explanation. But I will try to.